tests/versioned/apollo-federation/segments.test.js

```diff
@@ -0,0 +1,192 @@
+            }]
+          }]
+        }]
+      }]
+
+      t.segments(transaction.trace.root, expectedSegments)
+    })
+
+    executeQuery(serverUrl, query, (err, result) => {
+      t.error(err)
+      checkResult(t, result, () => {
+        t.end()
+      })
+    })
+  })
+
+  t.test('batch query should nest sub graphs under appropriate operations', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const booksQueryName = 'GetBooksForLibraries'
+    const booksQuery = `query ${booksQueryName} {
+      libraries {
+        branch
+        booksInStock {
+          isbn,
+          title,
+          author
+        }
+      }
+    }`
+
+    const magazineQueryName = 'GetMagazinesForLibraries'
+    const magazineQuery = `query ${magazineQueryName} {
+      libraries {
+        branch
+        magazinesInStock {
+          issue,
+          title
+        }
+      }
+    }`
+
+    const queries = [booksQuery, magazineQuery]
+
+    const libraryExternal = formatExternalSegment(t.context.libraryUrl)
+    const bookExternal = formatExternalSegment(t.context.bookUrl)
+    const magazineExternal = formatExternalSegment(t.context.magazineUrl)
+
+    helper.agent.on('transactionFinished', (transaction) => {
+      if (transaction.forceIgnore) {
+        return
+      }
+
+      const operationPart1 = `query/${booksQueryName}`
+      const expectedQuery1Name = `${operationPart1}`
+      const operationPart2 = `query/${magazineQueryName}`
+      const expectedQuery2Name = `${operationPart2}`
+
+      const batchTransactionPrefix = `${TRANSACTION_PREFIX}//batch`
+
+      const expectedSegments = [{
+        name: `${batchTransactionPrefix}/${expectedQuery1Name}/${expectedQuery2Name}`,
+        children: [{
+          name: 'Expressjs/Router: /',
+          children: [{
+            name: 'Nodejs/Middleware/Expressjs/<anonymous>',
+            children: [
+              {
+                name: `${OPERATION_PREFIX}/${operationPart1}`,
+                children: [
+                  { name: libraryExternal },
+                  { name: bookExternal }
+                ]
+              },
+              {
+                name: `${OPERATION_PREFIX}/${operationPart2}`,
+                children: [
+                  { name: libraryExternal },
+                  { name: magazineExternal }
+                ]
+              }
+            ]
+          }]
+        }]
+      }]
+
+      t.segments(transaction.trace.root, expectedSegments)
+    })
+
+    executeQueryBatch(serverUrl, queries, (err, result) => {
+      t.error(err)
+      checkResult(t, result, () => {
+        t.equal(result.length, 2)
+
+        t.end()
+      })
+    })
+  })
+}
+
+/**
+ * Verify we didn't break anything outright and
+ * test is setup correctly for functioning calls.
+ */
+ function checkResult(t, result, callback) {
+  t.ok(result)
+
+  if (result.errors) {
+    result.errors.forEach((error) => {
+      t.error(error)
+    })
+  }
+
+  setImmediate(callback)
+}
```

Comment:
This is getting enough uses now it probably could get extracted into its own module for reuse. I didn't tackle with this PR but likely will do so (or someone else can) later.